pkg/controller/controlplane/valuesprovider.go

```diff
@@ -426,15 +420,9 @@ func (vp *valuesProvider) GetStorageClassesChartValues(
 		}
 	}
 
-	k8sVersionLessThan121, err := version.CompareVersions(cluster.Shoot.Spec.Kubernetes.Version, "<", azure.CSIMigrationKubernetesVersion)
-	if err != nil {
-		return nil, err
-	}
-
 	values := map[string]interface{}{
-		"useLegacyProvisioner": k8sVersionLessThan121,
```

Comment:
Can't we just drop this value as this will always be false from now on. With that we can also adapt the files 
[charts/internal/shoot-storageclasses/templates/storageclasses.yaml](https://github.com/gardener/gardener-extension-provider-azure/blob/fddab7c19d8816ff5cc9cd381909fbd7a67a83d4/charts/internal/shoot-storageclasses/templates/storageclasses.yaml) and remove [charts/internal/shoot-storageclasses/templates/storageclasses_legacy.yaml](https://github.com/gardener/gardener-extension-provider-azure/blob/fddab7c19d8816ff5cc9cd381909fbd7a67a83d4/charts/internal/shoot-storageclasses/templates/storageclasses_legacy.yaml)

Reply:
I think so.